src/webchat-ui/components/WebchatUI.tsx

```diff
@@ -1,871 +1,938 @@
+												/>
+											)}
+										</WebchatRoot>
+									)}
+								{!disableToggleButton && (
+									<div>
+										{
+											// Show the message teaser if there is a last bot message and the webchat is closed
+											lastUnseenMessageText && (
+												<UnreadMessagePreview
+													className="webchat-unread-message-preview"
+													onClick={onToggle}
+													aria-live="polite"
+												>
+													<span className="sr-only">
+														New message preview
+													</span>
+													{lastUnseenMessageText}
+												</UnreadMessagePreview>
+											)
+										}
+										{isDisabled ? (
+											<div
+												title={getDisabledMessage()}
+												tabIndex={-1}
+												aria-disabled
+											>
+												<FABDisabled
+													data-cognigy-webchat-toggle
+													{...webchatToggleProps}
+													type="button"
+													className="webchat-toggle-button-disabled"
+													aria-label={getDisabledMessage()}
+													ref={this.chatToggleButtonRef}
+													disabled
+												>
+													<ChatIcon />
+												</FABDisabled>
+											</div>
+										) : (
+											<FAB
+												data-cognigy-webchat-toggle
+												onClick={onToggle}
+												{...webchatToggleProps}
+												type="button"
+												className="webchat-toggle-button"
+												aria-label={
+													open ? "Close chat" : openChatAriaLabel()
+												}
+												ref={this.chatToggleButtonRef}
+											>
+												{open ? <CollapseIcon /> : <ChatIcon />}
+												{config.settings.enableUnreadMessageBadge ? (
+													<Badge
+														content={unseenMessages.length}
+														className="webchat-unread-message-badge"
+														aria-label={`${unseenMessages.length} unread messages`}
+													/>
+												) : null}
+											</FAB>
+										)}
+									</div>
+								)}
+							</CacheProvider>
+						</WebchatWrapper>
+					</>
+				</ThemeProvider>
+			</>
+		);
+	}
+
+	renderRegularLayout() {
+		const {
+			currentSession,
+			config,
+			messages,
+			scrollToPosition,
+			lastScrolledPosition,
+			onSetScrollToPosition,
+			onSetLastScrolledPosition,
+			showHomeScreen,
+			onSetShowHomeScreen,
+			showPrevConversations,
+			onSetShowPrevConversations,
+			onSwitchSession,
+			onClose,
+			onEmitAnalytics,
+		} = this.props;
+
+		if (showHomeScreen)
+			return (
+				<HomeScreen
+					showHomeScreen={showHomeScreen}
+					onSetShowHomeScreen={onSetShowHomeScreen}
+					onSwitchSession={onSwitchSession}
+					currentSession={currentSession}
+					onSetShowPrevConversations={onSetShowPrevConversations}
+					onClose={onClose}
+					config={config}
+					onEmitAnalytics={onEmitAnalytics}
+				/>
+			);
+
+		const handleOnClose = () => {
+			onSetShowPrevConversations(false);
+			onSetShowHomeScreen(true);
+		};
+
+		const handleOnGoBack = () => {
```

Comment:
This way the back arrow is always going to HomeScreen. This is fine for the close icon, but this should be back to the previous view. For example: if I reach the chat from PrevConversationList, as user I expect to to be back on list.

Reply:
Actually close icon should always close the widget, not go back to home. I will check.

Reply:
I mean the arrow back. The close button behavior is fine to me. But the arrow back should go back to the previous screen, not just to Home Screen.

Reply:
This is out of scope. We will implement a navigation solution later, as per your suggestion earlier.